### PR TITLE
chore: remove some unneccessary abstractions

### PR DESCRIPTION
**Pull request by rockstar**
This patch just removes a few functions that were unneccessary, either because
they are a single, straightforward line of code, or because they were function
abstractions only ever called once. Most of this is carry over from the old,
hand rolled lsp, and this cleans the code up quite a bit.

### CODE REVIEW

**Review by asalem1 — COMMENTED** [review]
I'll be totally honest, I have no idea what this is doing. I'd love a line-by-line walkthrough of the diffs to better understand why the changes are being made

File: src/server/mod.rs [hunk 1]
```diff
@@ -79,44 +75,40 @@ fn replace_string_in_range(
...
-                walk::Node::Package(_) | walk::Node::File(_)
-                    if is_scope(name, *n) =>
-                {
-                    Some(n)
+                walk::Node::Package(_) | walk::Node::File(_) => {
+                    let mut visitor =
+                        semantic::DefinitionFinderVisitor::new(
+                            name.clone(),
+                        );
+                    walk::walk(&mut visitor, *n);
```
Comment (wiedld) [inline src/server/mod.rs:105]: Confirming what I think is going on:
* This is called when we already know the symbol (name) to lookup.
  * called per 1 symbol 
  * define a single/new `semantic::DefinitionFinderVisitor`.
* Start walk at the subtree roots (a package or file).
  * the 3 callers (rename, document_highlight, and references) all start at the root.
* The leaves:
  * a.k.a. VariableAssgn, BuiltinStmt, FunctionParameter.
  *  Stop the walk, and adds the found symbol if matches.

.

If this^^ is correct, then I don't understand what happens with variable shadowing:
* the namespace lookup starts in global scope, and then goes down the subtree.
* what if I am in an inner scope, and lookup a symbol (`find_references`) from the inner scope?
* Flux allows for shadowing: https://docs.influxdata.com/flux/v0.x/spec/assignment-scope/#variable-assignment. So think I am missing something here. 🤷🏼

File: src/server/mod.rs [hunk 2]
```diff
@@ -726,12 +694,16 @@ impl LanguageServer for LspServer {
...
 
-        let node = find_node(
-            walk::Node::Package(&pkg),
+        let mut visitor = semantic::NodeFinderVisitor::new(
             params.text_document_position.position,
         );
+        flux::semantic::walk::walk(
+            &mut visitor,
+            walk::Node::Package(&pkg),
+        );
```
Comment (wiedld) [inline src/server/mod.rs:703]: Assume this decorates/mutates the AST with semantic information?

**Review by wiedld — COMMENTED** [review]
Made comments without it being requested changes. Since I may be missing something.

In the end there may be no changes requested...but I want to make sure a few choices are intentional (or my misunderstanding). 😸

**Review by wiedld — APPROVED** [review]
Approving because is a functionally equivalent change.

Leaving the comments, since it highlights existing tech debt to be handled at some future point. (Note that the tech debt is contained in the choice of walking pattern used here, and doesn't cause compensatory tech debt to be added elsewhere.)

File: src/server/mod.rs [hunk 3]
```diff
@@ -79,44 +75,40 @@ fn replace_string_in_range(
...
-fn is_scope(name: &Symbol, n: walk::Node<'_>) -> bool {
-    let mut dvisitor =
-        semantic::DefinitionFinderVisitor::new(name.clone());
-    walk::walk(&mut dvisitor, n);
-
-    dvisitor.node.is_some()
-}
-
-fn find_references(
+fn find_references<'a>(
```
[the remark this answers is not in the document]
Comment (rockstar) [reply src/server/mod.rs:78]: > So this change will impact existing features and feature tests.

Not really, no. All it does is remove the use of `NodeFinderResult` and uses the values that were already a proxy. No functionality is being changed here; we're just removing some indirection so we can see what the actual interaction is. I'm _positive_ there are bugs here; I just didn't plan to address them here.

`FunctionExpr` comes from anywhere a function is defined, whether that's a function declaration or a parameter passed to `map`/`filter` etc. When you say "system functions", if the function is pure flux, than it's a `FunctionExpr`, but if it's builtin, that's a `BuiltinStmt`.

File: src/server/mod.rs (hunk 1 above)
Comment (rockstar) [reply src/server/mod.rs:105]: I _suspect_ when we shadow variables, we get some bug. I'm not really in a place to verify that, but it wouldn't surprise me. Most of our functions don't do a very good job of taking local scope into account (see #98 #452). As this is strictly a refactor, I'm prioritizing readability here. If you do find a bug, please file an issue about it.

File: src/server/mod.rs (hunk 2 above)
Comment (rockstar) [reply src/server/mod.rs:703]: If you look at the source for `find_node`, what we've replaced here is the contents of that function. No actual change in functionality. In this case, we're using a visitor to find a specific node.

**Comment by rockstar** [comment]
There's some awkward history here. When we first started working on the lsp again, it had bitrotted a bit, and was also doing a lot of work that other projects had popped up to do much better, e.g. handling of the jsonrpc protocol. Much of this code is a direct port from that old version of the lsp (what is essentially anything older than `0.6.x`). It's not very clean, maybe overly abstracted (as demonstrated here), and needs some hefty refactoring. This patch is an example of how we're dealing with that: if have a garden full of weeds, pick two or three every time you walk past. Hopefully, you walk past enough that you're picking them faster than they grow, and eventually you have no more weeds.

File: src/server/mod.rs (hunk 1 above)
Comment (wiedld) [reply src/server/mod.rs:105]: Tested, wrote issue ticket:
https://github.com/influxdata/flux-lsp/issues/479

**Pull request merged**